Automattic/jetpack
# Search: allow hiding content

## Proposed changes:

This change allows customizing which fields are highlighted in search results. Some users don't want to show parts of the content that match the query by default, especially if some of the content is behind a paywall/membership.

### Other information:


## Jetpack product discussion
p1724318175328299-slack-C02ME06LF

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

Add a snippet limiting highlighted fields:

```
function jp_filter_query( $options ) {
    $options['highlightFields'] = array( 'title' );
    return $options;
}
add_filter( 'jetpack_instant_search_options', 'jp_filter_query' );
```

search results should no longer show content, only title. It should be tested with all three search result formats (minimal, standard, product).

You can also test this with an excerpt field, but only if you apply D159359-code at the same time.
